skellytracker/trackers/openpose_tracker/openpose_recorder.py

```diff
@@ -0,0 +1,106 @@
+import json
+from typing import Dict, Union
+import numpy as np
+from pathlib import Path
+import re
+from tqdm import tqdm
+
+from skellytracker.trackers.base_tracker.base_recorder import BaseCumulativeRecorder
+from skellytracker.trackers.openpose_tracker.openpose_model_info import (
+    OpenPoseModelInfo,
+)
+
+
+class OpenPoseRecorder(BaseCumulativeRecorder):
+    def __init__(
+        self,
+        json_directory_path: Union[Path, str],
+        track_hands: bool = False,
+        track_faces: bool = False,
+    ):
+        super().__init__()
+        self.json_directory_path = Path(json_directory_path)
+        self.track_hands = track_hands
+        self.track_faces = track_faces
+
+    def extract_frame_index(self, filename: str) -> Union[int, None]:
+        """Extract the numeric part indicating the frame index from the filename."""
+        match = re.search(r"_(\d{12})_keypoints", filename)
+        return int(match.group(1)) if match else None
+
+    def parse_openpose_jsons(self, json_directory: Union[Path, str]) -> np.ndarray:
+        # Remove the iteration over subdirectories and focus on a single directory
+        json_directory = Path(json_directory)
+        files = list(Path(json_directory).glob("*.json"))
+        num_frames = len(files)
+        frame_indices = [self.extract_frame_index(f.name) for f in files]
+        frame_indices.sort()
+
+        num_markers = OpenPoseModelInfo.num_tracked_points
+        if self.track_hands:
+            num_markers += OpenPoseModelInfo.hand_markers * 2
+        if self.track_faces:
+            num_markers += OpenPoseModelInfo.face_markers
+
+        # Initialize a single camera array since we're only processing one video at a time
+        data_array = np.full((num_frames, num_markers, 3), np.nan)
+
+        # Process each JSON file in the directory
+        for file_index, json_file in enumerate(
+            tqdm(files, desc=f"Processing {json_directory.name} JSONs")
+        ):
+            with open(json_file) as f:
+                data = json.load(f)
+
+            if data["people"]:
+                keypoints = self.extract_keypoints(data["people"][0])
+                data_array[frame_indices[file_index], :, :] = keypoints
+
+        return data_array
+
+    def extract_keypoints(
+        self, person_data: Dict[str, np.ndarray]
+    ) -> (
+        np.ndarray
+    ):  # TODO: type hint person_data - is this an ndarray yet or something else?
+        """Extract and organize keypoints from person data."""
+
+        body_markers = OpenPoseModelInfo.num_tracked_points
+        hand_markers = OpenPoseModelInfo.hand_markers
+        face_markers = OpenPoseModelInfo.face_markers
+        # Initialize a full array of NaNs for keypoints
+        keypoints_array = np.full(
```

Comment:
I did some work on this class to handle the optional hand + face data better. Do we want to always return the same size array for data output (potentially with nans for hand and face data), or do we want to only include the data being tracked?

Reply:
Thanks for doing that, I was gonna tackle that sometime this week so it saves me the trouble. That's a good question, and I'm not sure I have a firm opinion. If we return the same size output, will the splitting/exporting functions down the line essentially create face/hands csvs that are empty? 

Oh also I just realized, if all those emptys go into skellyforge, we may get some completely hands and face data if it tries to interpolate those completely empty arrays.

Reply:
From a general perspective I support only saving the data we have. Both to reduce wasting unnecessary space, and also to avoid errors like the kind you mention with skellyforge.

However, it may make some of the checks in freemocap more difficult if it's not easy to determine the expected size of the array. We would have to add some method of checking (on the freemocap side) what flags were used to process in order to verify the array is a valid shape. That will be important for multiperson work as well.